### DISCUSSION TITLE
Upload speed seems slow on bot only?

### DISCUSSION THREAD

**Discussion by erickyun** [anasty17/mirror-leech-telegram-bot#1350] [Q&A]
Hi,

So I have 1 gbps upload/download speed.
I did a speedtest-cli speed test result were:
Download: 916.48 Mbit/s
Upload: 1143.44 Mbit/s

But with the bot I'm always getting `20.57MB` max upload speed?

My Download speed is fine most times btw `98.40MB`.
I'm only having this problem with upload speed?
Would appreciate if someone could help me please?

**Comment by anasty17** [comment]
Upload to what? To tg or to gd

**Reply by erickyun** [reply]
Both

**Reply by anasty17** [reply]
This depending on vps location and for leech upload so u should follow repo requirements updates. It's an update for pyrogram to avoid network issues. So mainly no code issue and wait for telethon many things will change.

**Reply by erickyun** [reply]
> This depending on vps location and for leech upload so u should follow repo requirements updates. It's an update for pyrogram to avoid network issues. So mainly no code issue and wait for telethon many things will change.

I was more like wondering what settings I need to change? Because when I use aria /m I get 2 or 3MB speed But when I use /yl yt-dlp I get 80MB download speed!

**Reply by anasty17** [reply]
Idk bro but sure its not code issue.
Check if u added speed limit in aria2c.
Maybe low speed from site or because of ur server location

**Reply by erickyun** [reply]
I understand thank you for your reply

**Comment by dqanshi** [comment]
Bro check your torrent links if its dead its so slow
